slick/src/main/scala/akka/stream/alpakka/slick/javadsl/Slick.scala

```diff
@@ -221,4 +365,11 @@ object Slick {
   private def toDBIO[T](javaDml: JFunction[T, String]): T => DBIO[Int] = { t =>
     SQLActionBuilder(javaDml.asScala(t), SetParameter.SetUnit).asUpdate
   }
+
+  private def toDBIO[T](javaDml: JBiFunction[T, Connection, PreparedStatement]): T => DBIO[Int] = { t =>
+    SimpleJdbcAction { ctx =>
+      val statement = javaDml(t, ctx.connection)
+      if (statement != null) statement.executeUpdate() else 0
```

Comment:
This becomes a hidden `filter` feature, but makes sense as a safe-guard.

Reply:
Yep. I thought about NPE-safety and I haven’t come up with anything better like this 🤷‍♂️

Reply:
But it becomes a dangerous mix with the failure handling you suggested in the example...

Reply:
I removed this check, rewritten examples in docs, and added test with `recover` function.
Please, review.